File: src/sanpingz.com/tests/loop/loop.go
package main

import (
		"fmt"
		"time"
		)

func main() {
	N := 1000000000
	i := 0
	sum := 0
	start := time.Now().UnixNano()
	for {
		i++
		sum += i
		if i == N {
			break
		}
	}
	end := time.Now().UnixNano()
	fmt.Printf("(Go) Run time for %d loop: %0.2fms\n", N, float32(end-start)/1000000)
}

File: src/sanpingz.com/tests/loop/loop.c
#include<stdio.h>
#include<sys/time.h>

//struct timeval{
//	long int tv_sec;	// sec
//	long int tv_usec;	// usec
//}

int main(){
	long N = 1000000000;
	long sum = 0;
	long i = 0;
	struct timeval start, end;
	gettimeofday(&start, NULL);
	for (;;) {
		i++;
		sum += i;
		if (i == N) {
			break;
		}
	}
	gettimeofday(&end, NULL);
	printf("(C) Run time for %ld loop: %fms\n", N, 1000*(end.tv_sec-start.tv_sec)+(end.tv_usec-start.tv_usec)/1000.0);
	return 0;
}
